src/nashEquilibria.jl

```diff
@@ -0,0 +1,207 @@
+#=
+Tools for computing Nash equilibria of normal form games.
+
+Includes: Pre-processing, Lemke-Howson, Simple Search.
+
+Authors: Arnav Sood
+=#
+
+using NonNegLeastSquares
+include("Games.jl")
+include("normal_form_game.jl")
+
+"""
+Render a n-player normal form game non-degenerate by adding a suitable positive constant to each payoff (i.e., by ensuring each payoff is strictly greater than 0).
+
+##### Arguments
+
+- `g::NormalFormGame` : A NormalFormGame object.
+
+##### Returns
+
+- `g::NormalFormGame` : A NormalFormGame object.
+"""
+function unDegenerate!(g::NormalFormGame)
+    # Get the normalizing constant.
+    numPlayers = num_players(g)
+    minArray = Array(Real, numPlayers)
+    for i in 1:numPlayers
+         minArray[i] = minimum(g.players[i].payoff_array)
+    end
+    normalizer = abs(minimum(minArray)) + 1
+
+    # Alter the payoff matrices.
+    for i in 1:numPlayers
+        g.players[i].payoff_array += normalizer
+    end
+    return g
+end
+
+# This is currently very slow; i.e., each call removes one dominated strategy for one player.
+
+"""
+Simplify a n-player normal form game by the iterated removal of dominated strategies.
+
+##### Arguments
+
+- `g::NormalFormGame` : A NormalFormGame object.
+
+##### Returns
+
+- `g::NormalFormGame` : A NormalFormGame object.
+"""
+function unDominate!(g::NormalFormGame)
+    changeFlag = true
+    changesMade = false
+    while changeFlag == true
+        changeFlag = unDominate_Auxiliary!(g::NormalFormGame)
+        changesMade = changeFlag
+    end
+    return changesMade
+end
+
+function unDominate_Auxiliary!(g::NormalFormGame)
+    # Iterate over players, flagging strictly dominated strategies, and then eliminating.
+    changesMade = false
+    for i in 1:num_players(g)
+        player = g.players[i]
+        numActions = num_actions(player)
+        payoffMatrix = player.payoff_array
+        dominatedActions = Array(Integer,0)
+        for j in 1:numActions
+            currentAction = payoffMatrix[j,:]
+            for k in symdiff(numActions,j)
+                 otherAction = payoffMatrix[k,:]
+                 if size(otherAction.>=currentAction) == size(currentAction)
```

Comment:
Hm. I'm afraid I don't see why. Wouldn't it be the case that it's only true if otherAction is always at least as good as our current action? (It looks like I changed from strict domination to weak domination; I need to change that back).